daemon/container.go

```diff
@@ -1441,6 +1442,18 @@ func (container *Container) startLogging() error {
 			return err
 		}
 		l = dl
+	case "gelf":
+		metadata := gelf.GelfMetadata{
```

Comment:
I think you can pass whole `container` to your logger. So, we will have less moving parts outside logger.